File: backend/api/routes/promo.js
const express = require('express');
const router = express.Router();
const { 
  getPromos, 
  getPromoById, 
  createPromo, 
  updatePromo, 
  deletePromo 
} = require('../controllers/promoController');
const { uploadPromo } = require('../middleware/uploadMiddleware');
const { authMiddleware, roleMiddleware } = require('../middleware');

// Public routes
router.get('/', getPromos);
router.get('/:id', getPromoById);

// Admin routes
router.post('/', authMiddleware, roleMiddleware('admin'), uploadPromo, createPromo);
router.put('/:id', authMiddleware, roleMiddleware('admin'), uploadPromo, updatePromo);
router.patch('/:id', authMiddleware, roleMiddleware('admin'), uploadPromo, updatePromo);
router.delete('/:id', authMiddleware, roleMiddleware('admin'), deletePromo);

module.exports = router;

File: fastapi/src/routes/promo.py
from fastapi import APIRouter
from src.controllers.promo_controller import get_promos, get_promo_by_id

router = APIRouter()


@router.get("/", response_model=dict)
async def get_promos_route():
    return await get_promos()


@router.get("/{promo_id}", response_model=dict)
async def get_promo_route(promo_id: str):
    return await get_promo_by_id(promo_id)